### PR TITLE
Add lexer support for Teradata BTEQ dot-prefixed meta-commands

### LINKED ISSUE #1673: Add support for BTEQ commands in Teradata parsing
BTEQ has a number of meta commands (docs at https://docs.teradata.com/r/jmAxXLdiDu6NiyjT6hhk7g/TvACxJd5BGW6uUDX_l2O4g). These are commands that generally start with a `.` character.

I see support for meta commands in postgres has been provided (https://github.com/sqlfluff/sqlfluff/pull/1423) - where we simply ignore the commands as long as they are confined to a single line. Doing the same thing here would be just fine.

## Expected Behaviour

We should simply ignore commands that start with a dot-command.

## Observed Behaviour

```
echo ".RUN FILE=POSTING" | sqlfluff parse --dialect teradata -
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        bteq_statement:
[L:  1, P:  1]      |            dot:                                              '.'
[L:  1, P:  2]      |            bteq_key_word_segment:
[L:  1, P:  2]      |                keyword:                                      'RUN'
[L:  1, P:  5]      |            unparsable:                                       !! Expected: 'Nothing...'
[L:  1, P:  5]      |                whitespace:                                   ' '
[L:  1, P:  6]      |                raw:                                          'FILE'
[L:  1, P: 10]      |                raw:                                          '='
[L:  1, P: 11]      |                raw:                                          'POSTING'
[L:  1, P: 18]      |    newline:                                                  '\n'

==== parsing violations ====
L:   1 | P:   5 |  PRS | Line 1, Position 5: Found unparsable section: ' FILE=POSTING'
```

## Dialect

Teradata

## Version

sqlfluff==0.7.0

## Configuration

Default

### PR DESCRIPTION
**Pull request by DresdenGman**
Add a lexer matcher for Teradata BTEQ meta-commands (dot-prefixed commands like `.RUN FILE=POSTING`, `.QUIT`, `.IF`) so the parser can skip them as comments.\n\nThis follows the same pattern as the Postgres psql meta-command support added in #1423.\n\nFixes #1673

---
## Summary by cubic
Handle Teradata BTEQ dot-prefixed commands (`.RUN`, `.QUIT`, `.IF`) using the existing grammar so they’re treated as non-SQL and SQL on the same line is preserved. Remove the regex-based lexer matcher that swallowed trailing SQL; fixes #1673.

<sup>Written for commit c732f985f7aea4deb772308b1fd9fe3102741bbf. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/sqlfluff/sqlfluff/pull/8114?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by keraion** [comment]
Hi @DresdenGman thanks for the PR. Can you please include tests covering what you've changed? Thanks!

**Comment by DresdenGman** [comment]
Thanks for the prompt! I've added `bteq_lexer.sql` fixture covering BTEQ commands interleaved with SQL, including `table.column` syntax to verify no false matches. Also fixed `insert_lexer_matchers(before="code")` to `before="newline"` (caught by the fixture generator). Regenerated `bteq.yml` to match the new lexer behavior.

**Comment by DresdenGman** [comment]
Added 3 edge-case fixtures covering inline BTEQ+SQL on same line, mixed-case keywords, and BTEQ without semicolons — all with auto-generated YML. The existing bteq_lexer fixture already covered basic BTEQ consumption + mixed valid SQL.

File: test/fixtures/dialects/teradata/bteq.yml [hunk 1]
```diff
@@ -3,32 +3,5 @@
...
-      dot: .
-      bteq_key_word_segment:
-        keyword: RUN
-      keyword: FILE
-      comparison_operator:
-        raw_comparison_operator: '='
-      bteq_file_path:
-        naked_identifier: POSTING
+_hash: fdda373cd9cd649f82a9c5cf7ba9e290375c0ceae29477b0bad5a25f24a52ae3
+file: null
```
Comment (keraion) [inline test/fixtures/dialects/teradata/bteq.yml]: I'm not sure if this is the right answer here.

File: test/fixtures/dialects/teradata/bteq_inline.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+-- Edge cases: BTEQ commands mixed inline with SQL on same line.
+-- These verify the lexer correctly splits BTEQ comment from SQL.
+.IF errorcode > 0 then .quit 4; SELECT 1;
+.RUN FILE=POSTING SELECT a FROM t;
+.QUIT SELECT 1 + 2;
+.LOGOFF SELECT * FROM t;
+
+SELECT 1; .LOGOFF
```
Comment (keraion) [inline test/fixtures/dialects/teradata/bteq_inline.sql]: The yaml suggests it doesn't handle the SQL found on the same line after because of the `[^\r\n]*`. It been a few years since I've used Teradata and BTEQs so I don't recall the behavior of a BTEQ like this.

**Review by keraion — CHANGES_REQUESTED** [review]
I have some concerns with the change in approach from what was added in #8023. I think it might be good to have a second pass on what should be lexed here.

**Comment by DresdenGman** [comment]
Thanks for the thorough review — you were right about the inline SQL issue. The  regex was consuming SQL on the same line after BTEQ commands.

I've reverted to the grammar-based approach — removed the RegexLexer entirely and restored the existing  grammar (which already parses BTEQ tokens precisely and stops at SQL boundaries, as introduced in #8023). The  fixture now shows the parse tree with  segments again instead of .

Let me know if there are other concerns.

**Comment by keraion** [comment]
I'll go ahead and close this if there isn't anything else to do here. That last commit basically undid everything. Feel free to open a new PR if there is any additional work you feel is needed.
